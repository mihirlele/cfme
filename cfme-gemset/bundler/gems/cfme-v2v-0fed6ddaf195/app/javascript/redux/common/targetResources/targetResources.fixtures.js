import Immutable from 'seamless-immutable';

export const targetClusters = Immutable({
  name: 'clusters',
  count: 5,
  subcount: 2,
  subquery_count: 2,
  pages: 1,
  resources: [
    {
      href: 'http://localhost:3000/api/clusters/10000000000006',
      id: '1',
      name: 'RHV Cluster1',
      ems_id: '10000000000007',
      created_on: '2016-08-03T02:19:35Z',
      updated_on: '2017-02-26T02:33:43Z',
      uid_ems: 'domain-c7',
      ha_enabled: false,
      ha_admit_control: true,
      ha_max_failures: 1,
      drs_enabled: false,
      drs_automation_level: 'fullyAutomated',
      drs_migration_threshold: 3,
      last_perf_capture_on: null,
      ems_ref_obj: 'domain-c7',
      effective_cpu: 109356,
      effective_memory: 789977235456,
      ems_ref: 'domain-c7',
      type: null,
      ext_management_system: { emstype: 'rhevm' },
      v_parent_datacenter: 'Default'
    },
    {
      href: 'http://localhost:3000/api/clusters/10000000000007',
      id: '2',
      name: 'RHV Cluster2',
      ems_id: '10000000000008',
      created_on: '2016-08-03T02:19:35Z',
      updated_on: '2017-02-26T02:33:43Z',
      uid_ems: 'domain-c7',
      ha_enabled: false,
      ha_admit_control: true,
      ha_max_failures: 1,
      drs_enabled: false,
      drs_automation_level: 'fullyAutomated',
      drs_migration_threshold: 3,
      last_perf_capture_on: null,
      ems_ref_obj: 'domain-c7',
      effective_cpu: 109356,
      effective_memory: 789977235456,
      ems_ref: 'domain-c7',
      type: null,
      ext_management_system: { emstype: 'rhevm' },
      v_parent_datacenter: 'Default'
    }
  ]
});

export const cloudTenants = Immutable({
  resources: [
    {
      href: 'http://0.0.0.0:8080/api/cloud_tenants/42000000000003',
      id: '42000000000003',
      name: 'HR Intranet',
      description: '',
      enabled: true,
      ems_ref: '39f6ca6e7cdd40278d67394d8144c2e7',
      ems_id: '42000000000004',
      created_at: '2018-11-06T04:05:55Z',
      updated_at: '2018-11-06T04:05:55Z',
      type: 'ManageIQ::Providers::Openstack::CloudManager::CloudTenant',
      parent_id: null,
      ext_management_system: {
        id: '42000000000004',
        name: 'Brilliant OpenStack',
        created_on: '2018-11-06T02:03:55Z',
        updated_on: '2018-11-09T21:33:18Z',
        guid: 'e8dda556-c667-42fa-80ea-5b5d4bda02de',
        zone_id: '42000000000001',
        type: 'ManageIQ::Providers::Openstack::CloudManager',
        api_version: 'v3',
        uid_ems: 'default',
        host_default_vnc_port_start: null,
        host_default_vnc_port_end: null,
        provider_region: 'regionOne',
        last_refresh_error: null,
        last_refresh_date: '2018-11-09T21:33:18Z',
        provider_id: null,
        realm: null,
        tenant_id: '42000000000001',
        project: null,
        parent_ems_id: null,
        subscription: null,
        last_metrics_error: null,
        last_metrics_update_date: null,
        last_metrics_success_date: null,
        tenant_mapping_enabled: false,
        enabled: true,
        options: {},
        zone_before_pause_id: null
      }
    },
    {
      href: 'http://0.0.0.0:8080/api/cloud_tenants/42000000000002',
      id: '42000000000002',
      name: 'admin',
      description: 'admin tenant',
      enabled: true,
      ems_ref: '641a64a1a42d429f9606b345f328d306',
      ems_id: '42000000000004',
      created_at: '2018-11-06T02:04:54Z',
      updated_at: '2018-11-06T02:04:54Z',
      type: 'ManageIQ::Providers::Openstack::CloudManager::CloudTenant',
      parent_id: null,
      ext_management_system: {
        id: '42000000000004',
        name: 'Brilliant OpenStack',
        created_on: '2018-11-06T02:03:55Z',
        updated_on: '2018-11-09T21:33:18Z',
        guid: 'e8dda556-c667-42fa-80ea-5b5d4bda02de',
        zone_id: '42000000000001',
        type: 'ManageIQ::Providers::Openstack::CloudManager',
        api_version: 'v3',
        uid_ems: 'default',
        host_default_vnc_port_start: null,
        host_default_vnc_port_end: null,
        provider_region: 'regionOne',
        last_refresh_error: null,
        last_refresh_date: '2018-11-09T21:33:18Z',
        provider_id: null,
        realm: null,
        tenant_id: '42000000000001',
        project: null,
        parent_ems_id: null,
        subscription: null,
        last_metrics_error: null,
        last_metrics_update_date: null,
        last_metrics_success_date: null,
        tenant_mapping_enabled: false,
        enabled: true,
        options: {},
        zone_before_pause_id: null
      }
    }
  ]
});

export const requestTargetClustersData = {
  method: 'GET',
  fetchTargetClustersUrl: '/api/dummyProviders',
  response: { data: targetClusters }
};
