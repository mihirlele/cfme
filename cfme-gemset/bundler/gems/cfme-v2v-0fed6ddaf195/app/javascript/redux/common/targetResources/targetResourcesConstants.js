import { RHV, OPENSTACK } from '../../../common/constants';

export const FETCH_V2V_TARGET_CLUSTERS = 'FETCH_V2V_TARGET_CLUSTERS';

export const FETCH_TARGET_COMPUTE_URLS = {
  [RHV]:
    '/api/clusters?expand=resources' +
    '&attributes=ext_management_system.emstype,v_parent_datacenter,ext_management_system.name,hosts' +
    '&filter[]=ext_management_system.emstype=rhevm',
  [OPENSTACK]:
    '/api/cloud_tenants?expand=resources&filter[]=ext_management_system.type=ManageIQ::Providers::Openstack::CloudManager&attributes=ext_management_system.name,ext_management_system.id,vms'
};
