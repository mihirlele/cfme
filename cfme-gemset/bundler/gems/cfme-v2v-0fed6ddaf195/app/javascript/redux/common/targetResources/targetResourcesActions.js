import URI from 'urijs';
import API from '../../../common/API';

import { FETCH_V2V_TARGET_CLUSTERS } from './targetResourcesConstants';

const _getTargetClustersActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_TARGET_CLUSTERS,
    payload: API.get(url)
  });

export const fetchTargetClustersAction = url => {
  const uri = new URI(url);
  return _getTargetClustersActionCreator(uri.toString());
};
