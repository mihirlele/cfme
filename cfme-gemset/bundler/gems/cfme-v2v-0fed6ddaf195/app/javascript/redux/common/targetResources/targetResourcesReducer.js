import Immutable from 'seamless-immutable';

import { FETCH_V2V_TARGET_CLUSTERS } from './targetResourcesConstants';

export const initialState = Immutable({
  isFetchingTargetClusters: false,
  isRejectedTargetClusters: false,
  errorFetchingTargetClusters: null,
  targetClusters: []
});

export default (state = initialState, action) => {
  switch (action.type) {
    case `${FETCH_V2V_TARGET_CLUSTERS}_PENDING`:
      return state.set('isFetchingTargetClusters', true).set('isRejectedTargetClusters', false);
    case `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`:
      if (action.payload.data && action.payload.data.resources && action.payload.data.resources[0]) {
        return state
          .set('targetClusters', action.payload.data.resources)
          .set('isRejectedTargetClusters', false)
          .set('isFetchingTargetClusters', false);
      }
      return state
        .set('targetClusters', [])
        .set('isRejectedTargetClusters', false)
        .set('isFetchingTargetClusters', false);
    case `${FETCH_V2V_TARGET_CLUSTERS}_REJECTED`:
      return state
        .set('errorFetchingTargetClusters', action.payload)
        .set('isRejectedTargetClusters', true)
        .set('isFetchingTargetClusters', false);
    default:
      return state;
  }
};
