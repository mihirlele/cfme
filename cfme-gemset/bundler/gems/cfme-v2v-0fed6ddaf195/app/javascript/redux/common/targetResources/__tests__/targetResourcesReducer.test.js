import targetResourcesReducer, { initialState } from '../targetResourcesReducer';
import { FETCH_V2V_TARGET_CLUSTERS } from '../targetResourcesConstants';
import { targetClusters } from '../targetResources.fixtures';

test('sets default state', () => {
  const action = { type: '@@INIT' };
  const state = targetResourcesReducer(undefined, action);

  expect(state).toMatchSnapshot();
});

describe('fetching target clusters', () => {
  test('is pending', () => {
    const action = {
      type: `${FETCH_V2V_TARGET_CLUSTERS}_PENDING`
    };
    const prevState = initialState.set('isRejectedTargetClusters', true);
    const state = targetResourcesReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_TARGET_CLUSTERS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingTargetClusters', true);
    const state = targetResourcesReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  describe('is successful,', () => {
    let prevState;
    beforeEach(() => {
      prevState = initialState.set('isRejectedTargetClusters', true).set('isFetchingTargetClusters', true);
    });
    test('and there is data', () => {
      const { resources } = targetClusters;
      const payload = {
        data: {
          ...targetClusters,
          resources: [
            {
              ...resources
            }
          ]
        }
      };
      const action = {
        type: `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = targetResourcesReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });

    test('and there is no data', () => {
      const payload = {
        data: {
          ...targetClusters,
          resources: []
        }
      };
      const action = {
        type: `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = targetResourcesReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });
  });
});
