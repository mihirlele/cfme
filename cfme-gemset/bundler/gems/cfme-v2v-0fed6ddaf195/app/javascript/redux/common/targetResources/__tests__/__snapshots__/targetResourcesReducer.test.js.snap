// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetching target clusters is pending 1`] = `
Object {
  "errorFetchingTargetClusters": null,
  "isFetchingTargetClusters": true,
  "isRejectedTargetClusters": false,
  "targetClusters": Array [],
}
`;

exports[`fetching target clusters is rejected 1`] = `
Object {
  "errorFetchingTargetClusters": "error",
  "isFetchingTargetClusters": false,
  "isRejectedTargetClusters": true,
  "targetClusters": Array [],
}
`;

exports[`fetching target clusters is successful, and there is data 1`] = `
Object {
  "errorFetchingTargetClusters": null,
  "isFetchingTargetClusters": false,
  "isRejectedTargetClusters": false,
  "targetClusters": Array [
    Object {
      "0": Object {
        "created_on": "2016-08-03T02:19:35Z",
        "drs_automation_level": "fullyAutomated",
        "drs_enabled": false,
        "drs_migration_threshold": 3,
        "effective_cpu": 109356,
        "effective_memory": 789977235456,
        "ems_id": "10000000000007",
        "ems_ref": "domain-c7",
        "ems_ref_obj": "domain-c7",
        "ext_management_system": Object {
          "emstype": "rhevm",
        },
        "ha_admit_control": true,
        "ha_enabled": false,
        "ha_max_failures": 1,
        "href": "http://localhost:3000/api/clusters/10000000000006",
        "id": "1",
        "last_perf_capture_on": null,
        "name": "RHV Cluster1",
        "type": null,
        "uid_ems": "domain-c7",
        "updated_on": "2017-02-26T02:33:43Z",
        "v_parent_datacenter": "Default",
      },
      "1": Object {
        "created_on": "2016-08-03T02:19:35Z",
        "drs_automation_level": "fullyAutomated",
        "drs_enabled": false,
        "drs_migration_threshold": 3,
        "effective_cpu": 109356,
        "effective_memory": 789977235456,
        "ems_id": "10000000000008",
        "ems_ref": "domain-c7",
        "ems_ref_obj": "domain-c7",
        "ext_management_system": Object {
          "emstype": "rhevm",
        },
        "ha_admit_control": true,
        "ha_enabled": false,
        "ha_max_failures": 1,
        "href": "http://localhost:3000/api/clusters/10000000000007",
        "id": "2",
        "last_perf_capture_on": null,
        "name": "RHV Cluster2",
        "type": null,
        "uid_ems": "domain-c7",
        "updated_on": "2017-02-26T02:33:43Z",
        "v_parent_datacenter": "Default",
      },
    },
  ],
}
`;

exports[`fetching target clusters is successful, and there is no data 1`] = `
Object {
  "errorFetchingTargetClusters": null,
  "isFetchingTargetClusters": false,
  "isRejectedTargetClusters": false,
  "targetClusters": Array [],
}
`;

exports[`sets default state 1`] = `
Object {
  "errorFetchingTargetClusters": null,
  "isFetchingTargetClusters": false,
  "isRejectedTargetClusters": false,
  "targetClusters": Array [],
}
`;
